cli_helpers/tabular_output/output_formatter.py

```diff
@@ -139,7 +143,12 @@ def format_output(self, data, headers, format_name=None,
         (_, _preprocessors, formatter,
          fkwargs) = self._output_formats[format_name]
         fkwargs.update(kwargs)
-        column_types = self._get_column_types(data)
+        if column_types is None:
+            data = list(data)
+            column_types = self._get_column_types(data)
+            data = iter(data)
```

Comment:
Can't we just leave the data as a list? I don't know if we gain anything by converting it back to an iterator, do we?